### ISSUE TITLE
Ability to specify SSH user

### ISSUE THREAD

**Issue by tedivm** [ivaldi/ivaldi-rsnapshot#4]
Many systems have root for SSH by default as a security feature. Being able to specify a different user and have it use sudo for root access would be a great feature.

**Comment by frenkel** [comment]
What is the `rsnapshot.conf` syntax for such a setup? Do you have an example?

**Comment by tedivm** [comment]
```
backup username@hostname:/file/path
```

Just change [this line](https://github.com/ivaldi/ivaldi-rsnapshot/blob/master/manifests/master/node_definition.pp#L11) to make it configurable instead of hard coded.

**Comment by frenkel** [comment]
Will it use sudo automatically?

-----Oorspronkelijk bericht-----
Van: "Robert Hafner" notifications@github.com
Verzonden: ‎18-‎3-‎2015 18:29
Aan: "ivaldi/ivaldi-rsnapshot" ivaldi-rsnapshot@noreply.github.com
CC: "Frank Groeneveld" frank@ivaldi.nl
Onderwerp: Re: [ivaldi-rsnapshot] Ability to specify SSH user (#4)

backup username@hostname:/file/path
Just change this line to make it configurable instead of hard coded.

**Comment by tedivm** [comment]
Nope, you should add a line into the sudoers file to allow the user to use rsync-

```
backupuser ALL= NOPASSWD:/usr/bin/rsync
```

Alternatively you can also force it so rsync is only accessible as the sender, so it can't be used to overwrite files-

```
backupuser ALL= NOPASSWD:/usr/bin/rsync --server
```

You may have to do some testing with that last one though, as I'm not sure rsnapshot does the proper ordering of arguments there by default.

**Comment by frenkel** [comment]
Let me rephrase my question: will rsnapshot automatically know to run `sudo rsync ...` when it is not run as root? It doesn't seem so according to the man page. Would we need to override the rsync command in rsnapshot.conf?

**I'm all for implementing this by the way, just want to do it correctly right away.**

**Comment by tedivm** [comment]
It won't be automatic, no. There's a couple of ways to do this but the easiest will probably be to add a wrapper script to the remote systems and then using the "--rsync-path" option (which you can add into rsnapshot with the "rsync_long_args" option in the configuration file).

This wrapper can be as simple as-

``` shell
/usr/bin/sudo /usr/bin/rsync --server $@
```

or it can be as complicated [as this](http://technokracy.net/2011/01/07/root_sudo_rsnapshot/).

**Comment by frenkel** [comment]
I've implemented this in the `non_root` branch. Could you test it?
Currently there is one bug: I can't get it to work with `rsync --server` in `/etc/sudoers`, I have to leave off the argument. Do you know why?

**Comment by frenkel** [comment]
@tedivm did you manage to test this?

**Comment by frenkel** [comment]
I've merged this with master, as it seems to work great in production.
